### DISCUSSION TITLE
Removing Jobs on Startup dotnet core

### DISCUSSION THREAD

**Discussion by billrob** [quartznet/quartznet#1674] [Q&A]
I'm using `services.AddQuartz(config => { })` to add my jobs and triggers, but don't see how to remove them.  Previously when working directly with the scheduler, I would keep a list of all my current jobs, loop the jobs in the database and delete the ones that don't exist anymore.  Then loop the list again and add any new jobs to the scheduler.  Then `scheduler.Start()`.

I don't see how to pull that off with the Fluent registration.  I know this has to be a solved problem, but every "Hello World" demo doesn't deal with that complexity.

**Comment by lahma** [comment]
Yes, the API is a bit hard to use for dynamic use case, it caters more for the static case. You can still obtain reference to the `ISchedulerFactory` from DI container and then call the `GetScheduler` on it to refresh the data.

In general, I think you wouldn't need separate lists if you'd just just `IScheduler` interface to maintain the data (jobs and triggers) in database.
